include/dsn/dist/replication/client_ddl.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Microsoft Corporation
+ *
+ * -=- Robust Distributed System Nucleus (rDSN) -=-
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+/*
+ * Description:
+ *     ddl client interface
+ *
+ * Revision history:
+ *     2015-12-30, xiaotz, first version
+ */
+
+#include <cctype>
+#include <dsn/dist/replication.h>
+#include <dsn/dist/replication/replication.types.h>
+
+
+using namespace dsn::replication;
+
+namespace dsn{ namespace client{
+
+class client_ddl : public clientlet
+{
+public:
+    client_ddl(std::vector<dsn::rpc_address> meta_servers);
+
+    dsn::error_code create_app(const std::string& app_name, const std::string& app_type, int partition_count, int replica_count);
+
+    dsn::error_code drop_app(const std::string& app_name);
+
+private:
+    bool inline static valid_app_char(int c);
```

Comment:
I suggest you putting this helper method into cpp file, to make the header file clear.